test/integration/test_phone_number_lookup.py

```diff
@@ -0,0 +1,168 @@
+
+        self.assertEqual(get_lookup_status_response.request_id, create_lookup_response[0].request_id)
+        self.assertIs(type(get_lookup_status_response), LookupStatus)
+        
+        # Assert that each result is of type LookupResult
+        for i in range(len(get_lookup_status_response.result)):
+            self.assertIs(type(get_lookup_status_response.result[i]), LookupResult)
+
+        # Check the information for a Bandwidth TN
+        bw_lookup_result: LookupResult = get_lookup_status_response.result[0]
+        self.assertEqual(bw_lookup_result.response_code, 0)
+        self.assertIs(type(bw_lookup_result.message), str)
+        self.assertEqual(bw_lookup_result.e_164_format, os.environ['BW_NUMBER'])
+        self.assertIs(type(bw_lookup_result.formatted), str)
+        self.assertTrue(bw_lookup_result.country == "US" or bw_lookup_result.country == "Canada")
+        self.assertTrue(bw_lookup_result.line_type == "Mobile" or bw_lookup_result.line_type == "Fixed")
+        self.assertIn("Bandwidth", bw_lookup_result.line_provider)
+
+        # Check the information for a Verizon(true mobile) TN
+        vzw_lookup_result = get_lookup_status_response.result[1]
+        self.assertEqual(vzw_lookup_result.response_code, 0)
+        self.assertIs(type(vzw_lookup_result.message), str)
+        self.assertEqual(vzw_lookup_result.e_164_format, os.environ['VZW_NUMBER'])
+        self.assertIs(type(vzw_lookup_result.formatted), str)
+        self.assertTrue(vzw_lookup_result.country == "US" or vzw_lookup_result.country == "Canada")
+        self.assertTrue(vzw_lookup_result.line_type == "Mobile" or vzw_lookup_result.line_type == "Fixed")
+        self.assertIn("Verizon", vzw_lookup_result.line_provider)
+        self.assertIs(type(vzw_lookup_result.mobile_country_code), str)
+        self.assertIs(type(vzw_lookup_result.mobile_network_code), str)
+
+        # The only way to get a failed number is if the api call to the downstream service fails - so there is no way to force this in our testing currently
+        # check the failed_telephone_number list 
+        # self.assertIs(type(get_lookup_status_response.failed_telephone_numbers), list)
+        # self.assertIn(os.environ['BW_INVALID_TN_LOOKUP_NUMBER'], get_lookup_status_response.failed_telephone_numbers)
+
+    
+    def testFailedPhoneNumberLookup(self):
+        """Test Phone Number Lookup API with bad data to force an error"""
+        with self.assertRaises(bandwidth.ApiException) as context:
+            lookup_request = LookupRequest(
+                tns=[
+                    'not a number',
+                ],
+            )
+            self.api_instance.create_lookup(self.account_id, lookup_request)
+        
+        self.assertIs(type(context.exception.status), int)
+        self.assertIs(type(context.exception.body), str)
+
+        # initialize TnLookupRequestError model 
+        error = TnLookupRequestError(message=(json.loads(context.exception.body))['message'])
```

Comment:
May be I'm missing something we should just try to parse the JSON body to expected model

Reply:
[Refer to this example](https://stackoverflow.com/questions/6578986/how-to-convert-json-data-into-a-python-object):
```
import json
from types import SimpleNamespace

data = '{"name": "John Smith", "hometown": {"name": "New York", "id": 123}}'

# Parse JSON into an object with attributes corresponding to dict keys.
x = json.loads(data, object_hook=lambda d: SimpleNamespace(**d))
print(x.name, x.hometown.name, x.hometown.id)
```

Reply:
So thats what I had originally planned on doing - but the openapi generator ignores all of the exception definitions defined in the api spec and creates its own [ApiException](https://github.com/Bandwidth/python-sdk/blob/f1e96d62cded74edf87c4a335a116b633de99c85/bandwidth/exceptions.py#L99) that it throws on [any 400 response](https://github.com/Bandwidth/python-sdk/blob/ba6b1d8d91179be64d8f97e2cee373170ed7e509/bandwidth/rest.py#L230) (besides 401m 403, 404, or 500) 

So my reasoning here was to initialize the `TnLookupRequestError` model that is created by the generator and otherwise never used. I _doubt_ any user will go the extra mile to parse the JSON and initialize the error class if its unneeded - but wanted to try and get full coverage of all the models related to this API.